library/std/src/io/readbuf.rs

```diff
@@ -20,230 +21,267 @@ use crate::mem::MaybeUninit;
 /// [ filled |         unfilled         ]
 /// [    initialized    | uninitialized ]
 /// ```
-pub struct ReadBuf<'a> {
-    buf: &'a mut [MaybeUninit<u8>],
+///
+/// A `BorrowedBuf` is created around some existing data (or capacity for data) via a unique reference
+/// (`&mut`). The `BorrowedBuf` can be configured (e.g., using `clear` or `set_init`), but otherwise
+/// is read-only. To write into the buffer, use `unfilled` to create a `BorrowedCursor`. The cursor
+/// has write-only access to the unfilled portion of the buffer (you can think of it like a
+/// write-only iterator).
```

Comment:
FWIW, I find the "read-only" claim a little confusing but I'm not sure I have concrete suggestions - it seems odd to call something read only that's intended to be written to in short order (albeit through a different type). Kind of like calling `Mutex<Vec<T>>` read-only, in some sense.

Seems OK to leave as-is for now.